ssr/src/page/wallet/txn.rs

```diff
@@ -393,13 +393,37 @@ pub mod provider {
                     }
                     IndexOrLedger::Ledger(ledger) => {
                         let ledger = self.canisters.sns_ledger(*ledger).await;
+                        // let history = ledger
+                        //     .get_transactions(GetTransactionsRequest {
+                        //         start: start.into(),
+                        //         length: (end - start).into(),
+                        //     })
+                        //     .await?;
+                        // let list_end = history.log_length < (end - start);
+                        // Ok(PageEntry {
+                        //     data: history
+                        //         .transactions
+                        //         .into_iter()
+                        //         .enumerate()
+                        //         .filter_map(|(i, txn)| {
+                        //             let idx = (history.first_index.clone() + i).0.to_u64_digits();
+                        //             if idx.is_empty() {
+                        //                 None
+                        //             } else {
+                        //                 parse_transactions_ledger(txn, idx[0]).ok()
+                        //             }
+                        //         })
+                        //         .collect(),
+                        //     end: list_end,
+                        // })
+
                         let history = ledger
                             .get_transactions(GetTransactionsRequest {
-                                start: start.into(),
-                                length: (end - start).into(),
+                                start: 0u32.into(),
+                                length: 10000000u32.into(),
```

Comment:
is there no other way? The cycle cost would be really high for tokens with many txns

Reply:
We could technically just recursively call it (thats what they are doing in nns) but it would cost more i think because it will fetch the same number of  txns but with more canister calls

Reply:
are we fine with this tradeoff?

Reply:
im pretty sure fetching global txns for the token will take too long as usage ramps up
there's also a payload limit of 2 MiB from ICP.
either we need pagination or we discard the idea of latest to newest

Fetching all the txns for a token is simply not scalable